src/sasl/cyrsasl_external.erl

```diff
@@ -66,73 +76,150 @@ mech_step(#state{creds = Creds}, User) ->
         false ->
             {error, <<"not-authorized">>};
         true ->
-            do_mech_step(Creds, User)
+            NewCreds = maybe_add_auth_id(Creds, User),
+            do_mech_step(NewCreds)
     end.
-do_mech_step(Creds, User) ->
-    XmppAddrs = get_credentials(Creds, xmpp_addresses),
-    CommonName = get_credentials(Creds, common_name),
+
+%%%=============================================================================
+%%%  local functions
+%%%=============================================================================
+start_all_modules() ->
+    Modules = [M || H <- ?MYHOSTS, {mod, M} <- get_verification_list(H)],
+    case code:ensure_modules_loaded(Modules) of
+        {error, Error} -> error(Error);
+        _ -> ok
+    end,
+    lists:map(fun start_module/1, Modules).
```

Comment:
`lists:maps/2` is okej, but `lists:foreach/2` might be used, since from what I understand we don't care about result.

Reply:
not actual any more. start/stop interfaces were removed from `cyrsasl` behavior, so this code is removed as well.